cpp/src/groupby/sort/group_histogram.cu

```diff
@@ -57,6 +58,12 @@ std::unique_ptr<column> build_histogram(column_view const& values,
   auto [distinct_indices, distinct_counts] =
     cudf::reduction::detail::compute_row_frequencies(labeled_values, partial_counts, stream, mr);
 
+  // compute_row_frequencies does not provide stable ordering
+  thrust::sort_by_key(rmm::exec_policy(stream),
+                      distinct_indices->begin(),
+                      distinct_indices->end(),
+                      distinct_counts->mutable_view().begin<int64_t>());
```

Comment:
will this sort impact the groupby histogram performance?

Reply:
There is slight regression around 5% on small input ( < 10'000) , but overall improvement from hides this extra **sory_by_key** . The overall speed up is upto 30% even with this sorting.